src/VecSim/algorithms/hnsw/hnsw_factory.cpp

```diff
@@ -119,4 +120,22 @@ size_t EstimateElementSize(const HNSWParams *params) {
      */
     return size_meta_data + size_total_data_per_element;
 }
+
+VecSimIndex *NewTieredIndex(const TieredHNSWParams *params,
+                            std::shared_ptr<VecSimAllocator> allocator) {
+    if (params->hnswParams.type == VecSimType_FLOAT32) {
+        auto *hnsw_index =
+            NewIndex_ChooseMultiOrSingle<float, float>(&params->hnswParams, allocator);
+        return new (allocator) TieredHNSWIndex<float, float>(
+            hnsw_index, params->jobQueue, params->submitCb, params->memoryCtx, params->UpdateMemCb);
+    } else if (params->hnswParams.type == VecSimType_FLOAT64) {
+        auto *hnsw_index =
+            NewIndex_ChooseMultiOrSingle<double, double>(&params->hnswParams, allocator);
+        return new (allocator) TieredHNSWIndex<double, double>(
+            hnsw_index, params->jobQueue, params->submitCb, params->memoryCtx, params->UpdateMemCb);
```

Comment:
Why not just pass `&params->hnswParams` to the hnsw new function? this way we'll have to update both this and the HNSW new function when support for other types will be added